src/AkeneoPimClientBuilder.php

```diff
@@ -255,7 +266,7 @@ protected function setUp(Authentication $authentication): array
     {
         $uriGenerator = new UriGenerator($this->baseUri);
 
-        $httpClient = new HttpClient($this->getHttpClient(), $this->getRequestFactory(), $this->getStreamFactory());
+        $httpClient = new HttpClient($this->getHttpClient(), $this->getRequestFactory(), $this->getStreamFactory(), $this->options);
```

Comment:
The way you did, if a PHP client use wants to configure the User-Agent, he has to do the following code:

```
$clientBuilder = new AkeneoPimClientBuilder('http://localhost/');
$clientBuilder->setUserAgent('a_user_agent");
$client = $clientBuilder->buildAuthenticatedByPassword('client_id', 'secret', 'admin', 'admin');
```

I think it's not simple for him because he has to add a new line of code BEFORE the clientBuilder->build and AFTER the initialization of the AkeneoPimClientBuilder.

Proposition

What do you think of the following solutions?

```
$options = ['user-agent' => 'a_user_agent'];
$clientBuilder = new AkeneoPimClientBuilder('http://localhost/', $options);
$client = $clientBuilder->buildAuthenticatedByPassword('client_id', 'secret', 'admin', 'admin');
```

Or

```
$options = ['user-agent' => 'a_user_agent'];
$clientBuilder = new AkeneoPimClientBuilder('http://localhost/');
$client = $clientBuilder->buildAuthenticatedByPassword('client_id', 'secret', 'admin', 'admin', $options);
```

with the following AkeneoPimClientBuilder constructor:

```
public function __construct(string $baseUri, array $options = [])
{
	// check options parameters
	...
}

```
And you can use the `Symfony\Component\OptionsResolver\OptionsResolver `for checking options.